### PR TITLE
Change variable name properties and validation failure error message (connect #2474 and connect #2143)

### PR DESCRIPTION
**Pull request by kymni**
#### Before the PR (what is the issue or what needed to be done)
Variable name properties read as question ID and validation error message was hard coded
#### The solution
Rephrase properties and create UI string for validation error message
## Checklist
* [ ] Connect the issue
* [ ] Test plan
* [ ] Copyright header
* [ ] Code formatting
* [ ] Documentation

### CODE REVIEW

**Comment by stellanl** [comment]
It's easy to rename the field in the backend. But what about old data?
We could
 1. Save the QuestionDto.variableName to both Question.questionId as well as the new Question.variableName. 
 2. On a fetch use variableName, and if not set, use questionId.
 3. Release.
 4. Run a data script that copies questionId to variableName.
 5. Update code (like reporting engine) that reads/writes questionId.
 6. Remove questionId field.

**Comment by muloem** [comment]
@stellanl sounds like a plan.  I think we should cover all but perhaps we leave out the data migration bit.  It should be enough to always check whether there is a variableName and if not check whether there is a questionId and return it as variable name.  And in the saving always save to variableName.  At some point down the road we can run a scan on the questionId field and perhaps drop if its empty

File: Dashboard/app/js/lib/models/models.js [hunk 1]
```diff
@@ -389,7 +389,7 @@ FLOW.QuestionAnswer = FLOW.BaseModel.extend({
   collectionDate: DS.attr('number'),
   surveyInstanceId: DS.attr('number'),
   iteration: DS.attr('number'),
-  questionID: DS.attr('string'),
+  questionID: DS.attr('string'), //TODO should be number?
```
Comment (muloem) [inline Dashboard/app/js/lib/models/models.js]: Should be a number but is quite widely used so better we tackle in a separate issue.

File: GAE/src/com/gallatinsystems/survey/domain/Question.java [hunk 2]
```diff
@@ -62,6 +62,7 @@
     private Long questionGroupId;
     private Long surveyId;
     private String questionId;
```
Comment (muloem) [inline GAE/src/com/gallatinsystems/survey/domain/Question.java]: Perhaps we can mark this one as deprecated and put a note of the transformation to `variableName`.

File: Dashboard/app/js/lib/models/models.js (hunk 1 above)
Comment (stellanl) [reply Dashboard/app/js/lib/models/models.js]: That's why I made it a TODO, and not simply changed it :)

**Comment by kymni** [comment]
#2143 was already closed as it covered the front-end bit of changing question ID to variable name. Commits for changing variables, properties, and functions have been made against #2474 which is where the test plan will be

File: Dashboard/app/js/lib/models/models.js (hunk 1 above)
Comment (stellanl) [reply Dashboard/app/js/lib/models/models.js]: See #2516.

**Pull request merged**